packages/web3-core/src/web3_context.ts

```diff
@@ -2,21 +2,37 @@ import { Web3APISpec } from 'web3-common';
 import { Web3ConfigOptions, SupportedProviders } from './types';
 import { Web3Config } from './web3_config';
 import { Web3RequestManager } from './web3_request_manager';
+import { Web3SubscriptionConstructor } from './web3_subscriptions';
+import { Web3SubscriptionManager } from './web3_subscription_manager';
 
-export class Web3Context<API extends Web3APISpec> extends Web3Config {
+export class Web3Context<
+	API extends Web3APISpec,
+	RegisteredSubs extends {
+		[key: string]: Web3SubscriptionConstructor<API>;
+	} = {},
+> extends Web3Config {
 	public static readonly providers = Web3RequestManager.providers;
 	public static givenProvider?: SupportedProviders<never>;
 	public readonly providers = Web3RequestManager.providers;
 
 	public readonly requestManager: Web3RequestManager<API>;
+	public readonly subscriptionManager?: Web3SubscriptionManager<API, RegisteredSubs>;
 
 	public constructor(
 		provider: SupportedProviders<API> | string,
 		options?: Partial<Web3ConfigOptions>,
+		registeredSubscriptions?: RegisteredSubs,
```

Comment:
Is this for continuation of already subscribed subs?

Reply:
Not subscribed subscriptions, but available subscriptions to be subscribed from subscription manager.

Reply:
in that case why we need to pass available subscriptions via new context to new instance of subscription manager?

Reply:
Every instance of subscription manager needs to know which type of subscriptions is supported, so it can provide valid type safe interface for it.